src/api/transaction/sign.js

```diff
@@ -17,20 +18,16 @@ const validate = utils.common.validate;
  */
 const HASH_TX_ID = 0x54584E00; // 'TXN'
 
-function serialize(txJSON) {
-  return core.SerializedObject.from_json(txJSON);
-}
-
 function hashSerialization(serialized, prefix) {
-  return serialized.hash(prefix || HASH_TX_ID).to_hex();
-}
-
-function signingData(txJSON) {
-  return core.Transaction.from_json(txJSON).signingData().buffer;
+  const hexPrefix = prefix.toString(16).toUpperCase();
+  const toHash = hexPrefix + serialized;
+  return hashjs.sha512().update(new Buffer(toHash, 'hex')).digest('hex')
+    .toUpperCase().slice(0, 64);
```

Comment:
A sha512 in hex would be 128 characters long? And you're cutting it in half? A comment to this function would help with understanding